velox/functions/prestosql/MapEntries.cpp

```diff
@@ -51,7 +51,7 @@ class MapEntriesFunction : public exec::VectorFunction {
   static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
     // map(K,V) -> array(row(K,V))
     return {exec::FunctionSignatureBuilder()
-                .knownTypeVariable("K")
```

Comment:
Are we sure about this ? Maps cannot have null keys and Unknown's represent null values right ? Can you add an example query from Presto which using map_entries which works with unknown type ?  Thanks !

Reply:
Hmm I'm not getting an error using null keys, even if the value is non-null. Is this a bug on Presto? 

 SELECT MAP_ENTRIES(MAP(x,y)) FROM (VALUES (NULL,ARRAY[])) AS t(x,y);
 SELECT MAP_ENTRIES(MAP(x,y)) FROM (VALUES (NULL,ARRAY[3])) AS t(x,y);
 SELECT MAP_ENTRIES(MAP(x,y)) FROM (VALUES (NULL,ARRAY[3,4,5])) AS t(x,y);
 
 The result is NULL

Reply:
I can modify the signature to keep knownType as Key and Value, and add another signature for Unknown Key and Value. 

But surprised these queries are working if a map can't have null keys. Are there any other examples I should try on Presto?

Reply:
TIL: So unknown is supported in a map but null keys are not : map() -> map<unknown, unknown>() . 
I wanted to be sure about this, so thanks for digging into this.